doc/source/whatsnew/v1.2.0.rst

```diff
@@ -379,6 +379,7 @@ Other
 - Fixed metadata propagation in the :class:`Series.dt` accessor (:issue:`28283`)
 - Bug in :meth:`Series.transform` would give incorrect results or raise when the argument ``func`` was dictionary (:issue:`35811`)
 - Bug in :meth:`Index.union` behaving differently depending on whether operand is a :class:`Index` or other list-like (:issue:`36384`)
+- Bug in :meth:`pandas.core.construction.sanitize_array` in which passing a >1d array to the :class:`Series` constructor raises a bare ``Exception``. Now raises a ``ValueError`` (:issue:`35744`)
```

Comment:
Probably better to refer to the Series constructor since sanitize_array is an internal function. Also not sure if I'd call this a bug; maybe something like "Series constructor now raises a ValueError...".

Reply:
@dsaxton will update -- do you think it's appropriate to move to the Enhancements -> Other Enhancements section? or leave in bugs?

Reply:
I think it's okay here

Reply:
cool - updated